auth.go

```diff
@@ -478,7 +478,7 @@ func authenticateWithConfig(sc *snowflakeConn) error {
 	//var consentCacheIdToken = true
 
 	if sc.cfg.Authenticator == AuthTypeExternalBrowser {
-		if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
+		if (runtime.GOOS == "windows" || runtime.GOOS == "darwin") && sc.cfg.ClientStoreTemporaryCredential != ConfigBoolFalse {
```

Comment:
This logic looks little weird. when flag is already true (i.e. !false) then why to make it true. Please make it clean to set the values.